subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -916,14 +929,16 @@ int bt_mesh_sensor_cli_get(struct bt_mesh_sensor_cli *cli,
 		return err;
 	}
 
-	if (rsp && !sensor_data.type) {
-		return -ENODEV;
+	if (!rsp) {
+		return 0;
 	}
 
-	if (rsp) {
-		*rsp = *sensor_data.value;
+	if (!sensor_data.type) {
+		return -ENODEV;
 	}
 
+	*rsp = *sensor_data.value;
```

Comment:
Is there an existing bug here? This will only copy the first channel, but we'd like the response to contain all channels. Should be 
```c
memcpy(rsp, sensor_data.value, sizeof(struct sensor_value) * sensor->channel_count);
```

Reply:
`count` is set to 1 at line 923.

Reply:
Yes, but that's just the number of sensors.This is attempting to copy all channels in each sensor

Reply:
Oh, I see, you are right.